File: SVN_REPO/assign1/python/test/test_perfectNumber.py
import unittest
from src.perfectNumber import is_perfect_number_imperative, is_perfect_number_functional

class TestPerfectNumber(unittest.TestCase):

    def test_perfect_number(self):
        test_cases = [6, 28, 496, 8128]
        for number in test_cases:
            with self.subTest(number=number):
                self.assertTrue(is_perfect_number_imperative(number), f"{number} is a perfect number (imperative)")
                self.assertTrue(is_perfect_number_functional(number), f"{number} is a perfect number (functional)")

    def test_non_perfect_number(self):
        non_perfect_numbers = [78, 48, 1234, 8, 20, 9, 11]
        for number in non_perfect_numbers:
            with self.subTest(number=number):
                self.assertFalse(is_perfect_number_imperative(number), f"{number} is NOT a perfect number (imperative)")
                self.assertFalse(is_perfect_number_functional(number), f"{number} is NOT a perfect number (functional)")

    def test_number_less_than_or_equal_to_1(self):
        test_cases = [0, 1, -1, -100]
        for number in test_cases:
            with self.subTest(number=number):
                self.assertFalse(is_perfect_number_imperative(number), f"{number} should return False (imperative)")
                self.assertFalse(is_perfect_number_functional(number), f"{number} should return False (functional)")

if __name__ == '__main__':
    unittest.main()

File: SVN_REPO/assign1/ruby/test/test_perfectNumber.rb
require 'test/unit'
require_relative '../src/perfectNumber'

class TestPerfectNumber < Test::Unit::TestCase
  def test_perfect_number
    test_cases = [6, 28, 496, 8128]
    test_cases.each do |number|
      assert(is_perfect_number_imperative(number), "#{number} is a perfect number (imperative)")
      assert(is_perfect_number_functional(number), "#{number} is a perfect number (functional)")
    end
  end

  def test_non_perfect_number
    non_perfect_numbers = [78, 48, 1234, 8, 20, 9, 11]
    non_perfect_numbers.each do |number|
      assert(!is_perfect_number_imperative(number), "#{number} is NOT a perfect number (imperative)")
      assert(!is_perfect_number_functional(number), "#{number} is NOT a perfect number (functional)")
    end
  end

  def test_number_less_than_or_equal_to_1
    test_cases = [0, 1, -1, -100]
    test_cases.each do |number|
      assert(!is_perfect_number_imperative(number), "#{number} should return False (imperative)")
      assert(!is_perfect_number_functional(number), "#{number} should return False (functional)")
    end
  end
end
